pkg/odo/cli/delete/component.go

```diff
@@ -22,15 +40,20 @@ func NewComponentOptions() *ComponentOptions {
 }
 
 func (o *ComponentOptions) SetClientset(clientset *clientset.Clientset) {
+	o.clientset = clientset
 }
 
 func (o *ComponentOptions) Complete(cmdline cmdline.Cmdline, args []string) (err error) {
+	if o.name == "" {
+		createParameters := genericclioptions.NewCreateParameters(cmdline).NeedDevfile("")
+		o.Context, err = genericclioptions.New(createParameters)
+		return
```

Comment:
Even if the language supports this, it is a better practice to explicitely return the value, to make it easier to read:
```
return err
```